examples/README.md

```diff
@@ -0,0 +1,164 @@
+
+**Source Code**: [squares.rs](squares.rs)
+
+This is a repetition of the square drawing, but with added colors and an inner
+colored fill using the `set_fill_color`, `begin_fill` and `end_fill` methods.
+
+### Isometric Nested Cubes
+
+![nestedcubes](../docs/assets/images/examples/nestedcubes.gif)
+
+**Source Code**: [nestedcubes.rs](nestedcubes.rs)
+
+This one is quite interesting: it draws a hexagonal spiral with a regularly
+fading color until it makes it go back to the original one, but it slowly
+reveals a form of visual illusion showing nested concentric three-dimensional
+cubes isometrically projected onto the two-dimensional space of the canvas.
+
+### Randomly-colored Tornado
+
+![coloredridge](../docs/assets/images/examples/coloredridge.gif)
+
+**Source Code**: [coloredridge.rs](coloredridge.rs)
+
+This color frenzy is also nice to look at: the regular shape is generated using
+a gaussian function setting the pen size.
+
+### Heart Emoji
+
+![heart](../docs/assets/images/examples/heart.gif)
+
+**Source Code**: [heart.rs](heart.rs)
+
+Cute heart. The arcs are not as easy to draw as one could imagine at first.
+
+### Rust Logo
+
+![rust](../docs/assets/images/examples/rust.gif)
+
+**Source Code**: [rust.rs](rust.rs)
+
+The - approximate - Rust language logo using this crate!
+
+### Snowman
+
+![snowman](../docs/assets/images/examples/snowman.gif)
+
+**Source Code**: [snowman.rs](snowman.rs)
+
+A nice-looking snowman with all its common features.
+
+### Turtle
+
+![draw_turtle](../docs/assets/images/examples/draw_turtle.gif)
+
+**Source Code**: [draw_turtle.rs](draw_turtle.rs)
+
+A turtle drawn by... a turtle!
+
+## Level: Advanced
+### Heighway Dragon Curve
+
+![dragon](../docs/assets/images/examples/dragon.gif)
+
+**Source Code**: [dragon.rs](dragon.rs)
+
+Draws a [Heighway dragon curve](https://en.wikipedia.org/wiki/Dragon_curve).
+Such a self-similar fractal curve is generated by recursively unfolding a line
+by 90°: step 1 is a simple segment, step 2 is duplicating step 1's segment and
+rotating it by 90° along one of its extremities, which makes a new extremity:
+repeat the process using this one in order to generate the next step.
+
+### Square Fractal Tree
+
+![tree](../docs/assets/images/examples/tree.gif)
+
+**Source Code**: [tree.rs](tree.rs)
+
+Another tree-like fractal curve where the trunks or branches are squares rather
+than lines, with sizes chosen to make two squares' vertices coincide.
+
+### Von Koch Snowflake
+
+![snowflake](../docs/assets/images/examples/snowflake.gif)
+
+**Source Code**: [snowflake.rs](snowflake.rs)
+
+Draws a [Von Koch snowflake](https://en.wikipedia.org/wiki/Koch_snowflake). This
+uses yet another recursive algorithm: step 1 is an equilateral triangle, step 2
+is cutting each of its sides in three and replaces their middle parts with an
+equilateral triangle of adequate size but without a base side, step n + 1 is
+repeating the process for each side of step n.
```

Comment:
I don't think this description accurately describes the process going on in the GIF. (Or maybe I just don't understand it.) Could you take a look and try to adjust it? We can also remove the description of the algorithm and just leave the Wikipedia link. That would be sufficient as well. :)

Reply:
Well, I tried to describe the construction process of the sides: at each step, cut every side in three, remove the middle one and replace it with a spike with sides of the same size. Is it still gibberish?

Reply:
I think the problem is that when you watch the GIF run, it isn't clear that your description is what is happening. The turtle draws two sides of a triangle over and over again while rotation just the right amount to produce the snowflake. We can't see where it is "cutting each of its sides in three and replacing their middle parts with an equilateral triangle...". That's why I think it would be fine to just leave off the description of how it works. The reader can look at the source code and wikipedia page themselves and figure it out from there.

```suggestion
Draws a [Von Koch snowflake](https://en.wikipedia.org/wiki/Koch_snowflake).
```

Does that make sense? It's totally not your fault. It's hard to describe these kinds of algorithms when you're just looking at a GIF of them running.

Reply:
Ah. Yes you are right. I guess that one was easier to map to what's happening in the GIF. Having said that, given that my reasoning is inconsistent I no longer have a strong position about whether we keep it as is or change it. I'll leave it up to you to decide what is clearest. :)

We can always go back and keep improving these things later!

Reply:
I should note that the reason I'm so willing to push things like this to later is not because I don't care. I actually care a lot. It's because your work is already high enough quality that this really doesn't matter either way. It's not worth holding up a PR over something small like this when it's already good enough. We can always come back and make it better later. :)

Reply:
Haha, I won the match :trophy:! Then I guess I'll adjust the description in order to be a bit clearer. I think adding a sentence like `Note there is no "cutting" happening during the drawing, all the drawn sides are the same length, but the turning angles are computed recursively.` could help in this regard. I'll also apply the equivalent of your suggestion on the dragon curve section.